### ISSUE TITLE
Auto capture exceptions in asyncData

### ISSUE THREAD

**Issue by iliyaZelenko** [nuxt-community/sentry-module#184]
### What problem does this feature solve?

It will not be necessary to use `$sentry.captureException(error)` to catch an error everywhere as described [here](https://github.com/nuxt-community/sentry-module#usage-in-asyncdata), each error will be logged itself if it appears in `asyncData`.

### What does the proposed changes look like?

I want the error to be caught through this module, for example, in this code:

``` js
async asyncData ({ params }) {
  if (params.id === '123') throw Error('Test error')

  someFunctionThatCanThrowError()
}
```

That is, not to write like this:

``` js
async asyncData ({ params }) {
  if (params.id === '123') {
    $sentry.captureException(Error('Test error'))
  }

  try {
    someFunctionThatCanThrowError()
  } catch (e) {
    $sentry.captureException(e)
  }
}
```

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/sentry-module/issues/c97">#c97</a>)</em></sub></div>

**Comment by rchl** [comment]
I think the exception will be caught regardless.

The linked example is probably just an example of usage.

Do you actually have an issue with exceptions not being captured from `asyncData`?

**Comment by iliyaZelenko** [comment]
I was my mistake, in fact it seems to work.

I have another problem: the stacktrace in such an error looks very strange. I use souremaps, if an error occurs on the client side, then I can clearly see stacktrace.

Error in `asyncData`:

![image](https://user-images.githubusercontent.com/13103045/83246199-73f26880-a1aa-11ea-9d6e-6aa868650da2.png)

Error on client side: 

![image](https://user-images.githubusercontent.com/13103045/83246548-f7ac5500-a1aa-11ea-8bb7-520e99758f81.png)

As you can see, on client side the stacktrace is clear, I can understand where the error occurred, on the server side everything is different. Are there any ways to improve this?

**Comment by rchl** [comment]
Yes, I think I might have had issues with that too (don't get server-side errors often so didn't bother analyzing).

I think it's a matter of either:
 - wrong source map setting for the server build
 - something wrong with matching source-maps on Sentry (possibly due to the module using the wrong prefix)

I'd need to analyze this at some point but any help is welcome.

**Comment by iliyaZelenko** [comment]
This will very-very help in debugging errors, I have a lot of code on the server side and it is not possible to transfer it to the client side in order to finally see sourcemaps. I look forward to news on this topic, keep me posted. Thanks for the quick answers!

**Comment by BenjaminGK** [comment]
@rchl any updates on this issue? We are facing this issue as well.
We have a single sentry project where we send both node and JS issues to.
We are using the latest version of the module but even though we specify in the publishRelease section the release name to be 'project@versionNumber' I can see in the sentry release file generated in The nuxt folder shows a sha key or similar.

**Comment by rchl** [comment]
Doesn't seem like a related issue (not creating releases properly vs. not able to see proper stack traces on server).

Sounds like a configuration issue. Create a new issue or discussion with all the details.
